### PR TITLE
Further tidy up of the Native Value Index

### PR DESCRIPTION
**Pull request by adamretter**
This code mainly replaces a few sets of `int` constants in `org.exist.xquery.Constants` with `Enum` equivalents. This makes ensuring that the correct values are passed a compile time issue, and also makes debugging in-flight options of the Native Value Index much much easier.

Apart from the `int` -> `Enum` changes, there are no other major code changes.

### CODE REVIEW

File: src/org/exist/xquery/GeneralComparison.java [hunk 1]
```diff
@@ -1171,32 +1172,28 @@ protected void switchOperands()
         context.getProfiler().message( this, Profiler.OPTIMIZATIONS, "OPTIMIZATION", "Switching operands" );
 
         //Invert relation
-        switch( relation ) {
+        switch(relation) {
 
-            case Constants.GT: {
-                relation = Constants.LT;
+            case GT:
+                relation = Comparison.LT;
```
Comment (shabanovd) [inline src/org/exist/xquery/GeneralComparison.java]: remove `Comparison.` here too?

File: src/org/exist/xquery/parser/XQueryParser.java [hunk 2]
```diff
@@ -84,7 +84,7 @@ public boolean foundErrors() {
 	}
 
 	public String getErrorMessage() {
-		StringBuilder buf= new StringBuilder();
+		StringBuffer buf= new StringBuffer();
```
Comment (shabanovd) [inline src/org/exist/xquery/parser/XQueryParser.java]: why use `StringBuffer` here? It used by a single thread, no?

File: src/org/exist/xquery/parser/XQueryTreeParser.java [hunk 3]
```diff
@@ -80,7 +81,7 @@ public boolean foundErrors() {
 	}
 
 	public String getErrorMessage() {
-		StringBuilder buf= new StringBuilder();
+		StringBuffer buf= new StringBuffer();
```
Comment (shabanovd) [inline src/org/exist/xquery/parser/XQueryTreeParser.java]: why switch to `StringBuffer`?

File: src/org/exist/xquery/value/DurationValue.java [hunk 4]
```diff
@@ -362,7 +364,7 @@ public boolean effectiveBooleanValue() throws XPathException {
     
     public static boolean areReallyEqual(Duration duration1, Duration duration2) {
     	final boolean secondsEqual = zeroIfNull((BigDecimal)duration1.getField(DatatypeConstants.SECONDS)).compareTo(
```
Comment (shabanovd) [inline src/org/exist/xquery/value/DurationValue.java]: `zeroIfNull` can return `null`

File: src/org/exist/xquery/Constants.java [hunk 5]
```diff
@@ -143,7 +143,6 @@
 	 * Identity operators
 	 */
 	public final static int IS = 14;
-	public final static int ISNOT = 15;
```
Comment (shabanovd) [inline src/org/exist/xquery/Constants.java]: is it possible to reuse `15`?

File: src/org/exist/xquery/GeneralComparison.java (hunk 1 above)
Comment (adamretter) [inline src/org/exist/xquery/GeneralComparison.java]: No, that won't compile.

File: src/org/exist/xquery/parser/XQueryParser.java (hunk 2 above)
Comment (adamretter) [inline src/org/exist/xquery/parser/XQueryParser.java]: This was in the Antlr `XQuery.g` grammar which generates this Java file, I didn't change this previously. However, I just fixed it in a subsequent commit for you

File: src/org/exist/xquery/parser/XQueryTreeParser.java (hunk 3 above)
Comment (adamretter) [inline src/org/exist/xquery/parser/XQueryTreeParser.java]: This was in the Antlr `XQueryTree.g` grammar which generates this Java file, I didn't change this previously. However, I just fixed it in a subsequent commit for you

File: src/org/exist/xquery/value/DurationValue.java (hunk 4 above)
Comment (adamretter) [inline src/org/exist/xquery/value/DurationValue.java]: I haven't changed this in my commit. That is out of scope here, feel free to fix in a future PR.

**Comment by adamretter** [comment]
@shabanovd I hope I have now addressed your concerns?

File: src/org/exist/xquery/GeneralComparison.java (hunk 1 above)
Comment (shabanovd) [inline src/org/exist/xquery/GeneralComparison.java]: will static import solve that compile errors? You have mix of patterns: short form, for example  `LT` and long one - `Comparison.LT`. Personally, I prefer short form.

Comment (shabanovd) [inline src/org/exist/xquery/GeneralComparison.java]: that this should be fixed for all enums

File: src/org/exist/xquery/value/DurationValue.java (hunk 4 above)
Comment (shabanovd) [inline src/org/exist/xquery/value/DurationValue.java]: so, you see bug and simply ignore it. Ok.

**Comment by shabanovd** [comment]
I'm not agree with styling of this PR, but going to merge it because enums changes required for other tasks. Hope that comments will be addressed in near future

**Pull request merged**